examples/07_tutorials/ai-reco-of-recos/utils/openai_data_models.py

```diff
@@ -0,0 +1,315 @@
+    "API_VERSION": azure_openai_api_version
+}
+
+azure_o1_model_info = {
+    "RESOURCE": azure_openai_resource,
+    "KEY": azure_openai_key,
+    "MODEL": os.getenv('AZURE_OPENAI_MODEL_O1', 'o1'),
+    "API_VERSION": azure_openai_api_version
+}
+
+
+azure_o1_mini_model_info = {
+    "RESOURCE": azure_openai_resource,
+    "KEY": azure_openai_key,
+    "MODEL": os.getenv('AZURE_OPENAI_MODEL_O1_MINI', 'o1-mini'),
+    "API_VERSION": azure_openai_api_version
+}
+
+
+azure_o3_mini_model_info = {
+    "RESOURCE": azure_openai_resource,
+    "KEY": azure_openai_key,
+    "MODEL": os.getenv('AZURE_OPENAI_MODEL_O3_MINI', 'o3-mini'),
+    "API_VERSION": azure_openai_api_version
+}
+
+azure_o3_model_info = {
+    "RESOURCE": azure_openai_resource,
+    "KEY": azure_openai_key,
+    "MODEL": os.getenv('AZURE_OPENAI_MODEL_O3', 'o3'),
+    "API_VERSION": azure_openai_api_version
+}
+
+azure_o4_mini_model_info = {
+    "RESOURCE": azure_openai_resource,
+    "KEY": azure_openai_key,
+    "MODEL": os.getenv('AZURE_OPENAI_MODEL_O4_MINI', 'o4-mini'),
+    "API_VERSION": azure_openai_api_version
+}
+
+azure_ada_embedding_model_info = {
+    "RESOURCE": azure_openai_resource,
+    "KEY": azure_openai_key,
+    "MODEL": os.getenv('AZURE_OPENAI_MODEL_EMBEDDING_ADA', 'text-embedding-ada-002'),
+    "API_VERSION": azure_openai_api_version,
+    "DIMS": 1536
+}
+
+azure_small_embedding_model_info = {
+    "RESOURCE": azure_openai_resource,
+    "KEY": azure_openai_key,
+    "MODEL": os.getenv('AZURE_OPENAI_MODEL_EMBEDDING_SMALL', 'text-embedding-3-small'),
+    "API_VERSION": azure_openai_api_version,
+    "DIMS": 1536
+}
+
+azure_large_embedding_model_info = {
+    "RESOURCE": azure_openai_resource,
+    "KEY": azure_openai_key,
+    "MODEL": os.getenv('AZURE_OPENAI_MODEL_EMBEDDING_LARGE', 'text-embedding-3-large'),
+    "API_VERSION": azure_openai_api_version,
+    "DIMS": 3072
+}
+
+openai_gpt_45_model_info = {
+    "KEY": os.getenv('OPENAI_API_KEY'),
+    "MODEL": os.getenv('OPENAI_MODEL_4O')
+}
+
+openai_gpt_4o_model_info = {
+    "KEY": os.getenv('OPENAI_API_KEY'),
+    "MODEL": os.getenv('OPENAI_MODEL_4O')
+}
+
+openai_o1_model_info = {
+    "KEY": os.getenv('OPENAI_API_KEY'),
+    "MODEL": os.getenv('OPENAI_MODEL_O1')
+}
+
+openai_o1_mini_model_info = {
+    "KEY": os.getenv('OPENAI_API_KEY'),
+    "MODEL": os.getenv('OPENAI_MODEL_O1_MINI')
+}
+
+openai_o3_model_info = {
+    "KEY": os.getenv('OPENAI_API_KEY'),
+    "MODEL": os.getenv('OPENAI_MODEL_O3', 'o3')
+}
+
+openai_o3_mini_model_info = {
+    "KEY": os.getenv('OPENAI_API_KEY'),
+    "MODEL": os.getenv('OPENAI_MODEL_O3_MINI')
+}
+
+openai_o4_mini_model_info = {
+    "KEY": os.getenv('OPENAI_API_KEY'),
+    "MODEL": os.getenv('OPENAI_MODEL_O4_MINI')
+}
+
+openai_embedding_model_info = {
+    "KEY": os.getenv('OPENAI_API_KEY'),
+    "MODEL": os.getenv('OPENAI_MODEL_EMBEDDING'),
+    "DIMS": 3072 if os.getenv('AZURE_OPENAI_MODEL_EMBEDDING') == "text-embedding-3-large" else 1536
```

Comment:
Which models are covered by the `else` case here? Can we be certain that all their dimensions are `1536`?